### PR TITLE
Vanguard integration

### LINKED ISSUE #2008: SPIKE - Integration with ADFS Login
### Problem description
Investigate what we need to do to integrate with ADFS for single sign-on

### PR DESCRIPTION
**Pull request by nahidakbar**
### What this PR does

Fixes #2008 

This PR contains functionality necessary for connecting magda instances to vanguard through ws-fed protocol. 

How to test.

1. Make sure the metadata set is correct (has the right return url etc): https://magda-files.s3-ap-southeast-2.amazonaws.com/WSFED-ENV.xml
2. Goto https://cloud.thirdparty.authentication.business.gov.au/sts/claims/ and set user data
3. Goto login page, click on login via vanguard and then set third party integration test as login provider

For whoever picking this up, remaining stuff including private keys etc is here: https://drive.google.com/drive/folders/1vniUnRqOIfvyUy8nekDueITwEUUKzs96

### Checklist

-   [x] Unit tests aren't applicable
-   [x] I've updated CHANGES.md with what I changed.
-   [x] I've linked this PR to an issue in ZenHub (core dev team only)

### CODE REVIEW

File: magda-gateway/src/createOrGetUserToken.ts [hunk 1]
```diff
@@ -10,6 +10,7 @@ export default function createOrGetUserToken(
     profile: passport.Profile,
     source: string
 ): Promise<UserToken> {
+    console.log("USER", profileToUser(profile, source));
```
Comment (AlexGilleran) [inline magda-gateway/src/createOrGetUserToken.ts]: Probably shouldn't be logging this in prod.

File: magda-gateway/src/oauth2/vanguard.ts [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+export default function vanguard(options: VanguardOptions) {
+    const authorizationApi = options.authorizationApi;
+    const passport = options.passport;
+    const wsFedIdpUrl = options.wsFedIdpUrl;
+    const wsFedRealm = options.wsFedRealm;
+    const wsFedCertificate = options.wsFedCertificate;
+    const externalAuthHome = options.externalAuthHome;
+
+    // const loginBaseUrl = `${externalAuthHome}/login`;
```
Comment (AlexGilleran) [inline magda-gateway/src/oauth2/vanguard.ts]: Can get rid of this?

File: magda-gateway/package.json [hunk 3]
```diff
@@ -43,6 +43,7 @@
     "passport-facebook": "^2.0.0",
     "passport-google-oauth20": "^1.0.0",
     "passport-local": "^1.0.0",
+    "passport-wsfed-saml2": "^3.0.17",
```
Comment (AlexGilleran) [inline magda-gateway/package.json]: I think we might need to fork our version of this with the x509 and cryptiles problems patched out :(. No need for you to get to this before you go though.

Comment (nahidakbar) [reply magda-gateway/package.json]: https://github.com/auth0/passport-wsfed-saml2/pull/129

Comment (nahidakbar) [reply magda-gateway/package.json]: Will native x509 dependency cause problems? It just needs build tools for npm install.

Comment (AlexGilleran) [reply magda-gateway/package.json]: Apparently it breaks on node > 8?

Comment (AlexGilleran) [reply magda-gateway/package.json]: There's already a PR to get rid of it, but it's been sitting since October :(

**Pull request merged**